test_scripts/API/Navigation/SendLocation/020_SendLocation_boundary_by_parameters.lua

```diff
@@ -0,0 +1,271 @@
+local string500Char = string.rep("a", 500)
+local string255Char = string.rep("a", 255)
+local string200Char = string.rep("a", 200)
+
+local request_params = {
+    longitudeDegrees = 1.1,
+    latitudeDegrees = 1.1
+}
+
+local addressLinesArrayUpperBoundSize = {}
+for i=1,maxAdressLinesArraySize do
+    addressLinesArrayUpperBoundSize[i] = "some string"
+end
+local addressLinesArrayUpperBoundSizeValue = {}
+for i=1,maxAdressLinesArraySize do
+    addressLinesArrayUpperBoundSizeValue[i] = string500Char
+end
+
+local DateTimeParams =
+{
+    millisecond = {lower = 0, upper = 999},
+    second = {lower = 0, upper = 60},
+    minute = {lower = 0, upper = 59},
+    hour = {lower = 0, upper = 23},
+    day = {lower = 1, upper = 31},
+    month  = {name = "month", lower = 1, upper = 12},
+    --According to comfirmed information before 'should not be any min value specified',
+    -- so as min value specified negative value to check correct processing value not from positive range.
+    year = {lower = -1, upper = 4095},
+    tz_hour = {lower = -12, upper = 14},
+    tz_minute = {lower = 0, upper = 59},
+}
+
+local addressParams =
+{
+    "countryName",
+    "countryCode",
+    "postalCode",
+    "administrativeArea",
+    "subAdministrativeArea",
+    "locality",
+    "subLocality",
+    "thoroughfare",
+    "subThoroughfare",
+}
+
+local boundValues ={
+    longitudeDegrees = {lower = -180, upper = 180},
+    latitudeDegrees = {lower = -90, upper = 90},
+    locationName = {lower = "a", upper = string500Char},
+    locationDescription = {lower = "a", upper = string500Char},
+    phoneNumber = {lower = "a", upper = string500Char},
+    addressLines = {lower = {"a"}, upper = {string500Char}}
+}
+
+local locationImage = {lower = "a", upper = string255Char}
+
+local deliveryModeEnum = {
+    "PROMPT",
+    "DESTINATION",
+    "QUEUE"
+}
+
+--[[ Local Functions ]]
+local function sendLocation(parameter, value, self)
+    local paramters  = commonFunctions:cloneTable(request_params)
+
+    paramters[parameter] = value
+
+    local cid = self.mobileSession1:SendRPC("SendLocation", paramters)
+
+    paramters.appID = commonSendLocation.getHMIAppId()
+
+    EXPECT_HMICALL("Navigation.SendLocation", paramters)
+    :Do(function(_,data)
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+
+    self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+
+local function sendLocationAddress(subParam, value, self)
+    local paramters  = commonFunctions:cloneTable(request_params)
+
+    paramters["address"] = {}
+    paramters.address[subParam] = value
+
+    local cid = self.mobileSession1:SendRPC("SendLocation", paramters)
+
+    paramters.appID = commonSendLocation.getHMIAppId()
+
+    EXPECT_HMICALL("Navigation.SendLocation", paramters)
+    :Do(function(_,data)
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    :ValidIf (function(_,data)
```

Comment:
@GetmanetsIrina Please remove extra space

Reply:
Done in 4875635ea9ce242266c4f0d7bbaece1731a76faf

Reply:
@GetmanetsIrina Please also remove extra space in lines  187, 229, 249

Reply:
@dboltovskyi done in d0f8f738ed85818574f90acc1831e910b41ec179